Petshop-iOS/Petshop-iOS/Views/AboutUsView.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  AboutUsView.swift
+//  Petshop-iOS
+//
+//  Created by Matheus Ferreira on 30/06/24.
+//
+
+import UIKit
+
+class AboutUsView: UIView {
+    
+    private lazy var catDogImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.image = UIImage(named: "catdog")
+        imageView.contentMode = .scaleAspectFit
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        return imageView
+    }()
+    
+    private let cardView: CardView = {
+        let cardView = CardView()
+        cardView.translatesAutoresizingMaskIntoConstraints = false
+        return cardView
+    }()
+
+    private var hasAnimated = false
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupUI()
+        animateElements()
+        configureCardView()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("(init(coder:) has not been implemented")
+    }
+    
+    private func setupUI() {
+        backgroundColor = .blueColor
```

Comment:
Aqui algo parecido, ao invés de ser `.blueColor`, o atributo talvez devesse ser `.backgroundColor`, pois se amanhã mudar isso dará mais trabalho do que deveria. Poderia ficar algo do tipo `backgroundColor = .backgroundColor`.

Reply:
Isso não faz sentido, pois vai ter diferentes backgrounds durante o fluxo do app. Nomear blueColor de backgroundColor iria gerar uma confusão, fora que, o azul usado no background também é usado em outras partes da tela que não são background.

Reply:
Nesse caso podemos atribuir nomes específicos pra utilização: primaryColor, secondaryColor, accentColor, backgroundColor, textColor, titleColor, errorColor, sei la.. blueColor é ruim pq teremos pelo menos 2 tons de azul, entao.. que azul seria esse? Nosso background vai ser sempre o tom de branco.. O azul foi só na view de start. Nós vamos manter um padrão limite de cores, então ficaria interessante pensar numa estrutura de nomes genéricos mesmo. Apesar de termos varios backgrounds possiveis, seria só mencionar qual cor é: principal? secundária? texto principal? texto de destaque? de erro? Catalogue todas as cores que temos até o momento e veja aonde elas são usadas com frequencia, nomeie a partir daí pela funçao delas.

Reply:
sim, mas se definir como titleColor e a cor dos títulos mudaram futuramente não daria no mesmo? eu acho essas definições tão ruins quanto o nome da cor.

Reply:
Eu posso passar o nome da cor + onde está, como blueBackgroundColor. Mas aí também acho ruim, pois esse azul não tá sendo apenas usado no background. O que acha?

Reply:
A questão é legibilidade de código. A cor pode mudar mil vezes, mas a ideia de onde ela é aplicada, vai ser um padrão. Se o titulo mudar a cor, ainda estaremos falando do padrão da cor do título. Se amanha a gente muda de azul pra laranja, voce vai ter que alterar manualmente no seu código todo o `blueColor` para `orangeColor`para que a aplicacao nao tenha um `backgroundColor = .blueColor`que  estaria setando laranja.. confusão total.

https://primarytech.com/best-practices-for-naming-design-tokens-components-and-variables/

Reply:
Entendi, faz sentido.

Reply:
Não há problema se tivermos 2 variáveis com a mesma cor.. exemplo: `#42A2D9`
Se ela existe repetidas vezes no background mas também em textos, faz sentido separar pra que se futuramente alterarmos o padrão de fundo, não precise alterar o padrão do texto junto..
primaryBackgroundColor: `#42A2D9`, 
secondaryBackgroundColor: `#FFFFFF`

primaryTextColor: `#42A2D9`, 
secondaryTextColor: `#654321`

primaryButtonColor: `#42A2D9`, 
disabledButtonColor: `#999999`,

Reply:
@lechitz tu foi perfeito. Entendeu de fato o que eu quis dizer. Olhaaaaa o meu Jr pleno mandando bem!!! 

@MatheusFerreiraZx desculpa se eu falhei em ser claro suficiente e em te dar uma explicação melhor.

Reply:
@MaironCosta o bravo já virou pleno já, esse cara tá voando!